include/dali/core/tensor_shape.h

```diff
@@ -691,14 +730,16 @@ struct TensorListShape<DynamicDimensions>
         ndim(get_dim_from_uniform(sample_shapes)) {}
 
   TensorListShape(const std::vector<int64_t> &shapes, int sample_dim)
-      : Base(shapes, shapes.size() / sample_dim), ndim(sample_dim) {}
+      : Base(shapes, shapes.size() / (sample_dim == 0 ? 1 : sample_dim)),
```

Comment:
This constructor makes no sense for 0D. We should leave it crashing for 0D or remove altogether, if it's only used in tests.